File: app/src/main/java/im/bernier/petfinder/model/Shelter.java
package im.bernier.petfinder.model;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

/**
 * Created by Michael on 2016-10-29.
 */

@Root(name = "shelter", strict = false)
public class Shelter {

    @Element
    private String id;

    @Element
    private String name;

    @Element(required = false)
    private String address1;

    @Element(required = false)
    private String address2;

    @Element
    private String city;

    @Element
    private String state;

    @Element
    private String country;

    @Element
    private String zip;

    @Element
    private double latitude;

    @Element
    private double longitude;

    @Element(required = false)
    private String phone;

    @Element
    private String email;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getAddress1() {
        return address1;
    }

    public void setAddress1(String address1) {
        this.address1 = address1;
    }

    public String getAddress2() {
        return address2;
    }

    public void setAddress2(String address2) {
        this.address2 = address2;
    }

    @Override
    public String toString() {
        return "Shelter{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", address1='" + address1 + '\'' +
                ", address2='" + address2 + '\'' +
                '}';
    }
}

File: app/src/main/java/im/bernier/petfinder/model/Shelter.kt
/*
 *   This Source Code Form is subject to the terms of the Mozilla Public
 *   License, v. 2.0. If a copy of the MPL was not distributed with this
 *   file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *
 * If it is not possible or desirable to put the notice in a particular
 * file, then You may include the notice in a location (such as a LICENSE
 * file in a relevant directory) where a recipient would be likely to look
 * for such a notice.
 *
 * You may add additional accurate notices of copyright ownership.
 */

package im.bernier.petfinder.model

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

/**
 * Created by Michael on 2016-10-29.
 */

@Root(name = "shelter", strict = false)
data class Shelter(@field:Element var id: String? = null, @field:Element var name: String? = null, @field:Element(required = false) var address1: String? = null, @field:Element(required = false) var address2: String? = null, @field:Element var city: String? = null, @field:Element var state: String? = null, @field:Element var country: String? = null, @field:Element var zip: String? = null, @field:Element var latitude: Double = 0.toDouble(), @field:Element var longitude: Double = 0.toDouble(), @field:Element(required = false) var phone: String? = null, @field:Element var email: String? = null)

